internal/controller/gateway_mutator_test.go

```diff
@@ -252,6 +276,50 @@ func TestGatewayMutator_mutatePod(t *testing.T) {
 						require.NoError(t, err)
 					}
 
+					// Create Gateway and GatewayConfig for GatewayConfig test case
+					if tt.name == "with GatewayConfig" {
```

Comment:
Instead of relying on the test name, which makes it harder to evolve these tests, just add a property to the test case: `gatewayConfig *aigv1a1.GatewayConfig`, and if it is set, then create it with the fake client. You can extract this GW config into a package variable to make it easier to reference in the test if most of its information is static.